### DISCUSSION TITLE
Filter doesn't work with Boolean type with custom values

### DISCUSSION THREAD

**Discussion by tommiekn** [laravel/nova-issues#5031] [Q&A] [answered]
- Laravel Version: 9.38.0
- Nova Version: 4.18.2
- PHP Version: 8.1.11

### Description:
Using a Boolean type field with custom values, Nova is unable to filter by true value.

### Detailed steps to reproduce the issue on a fresh Nova installation:
Example Model:

` protected $casts = [
        'variable' => Boolswitch::class
   ]`

Boolswitch class is a legacy class as some bool we use has different true/false values.

In this example, the database is saved with on/off values.

```
class Boolswitch implements CastsAttributes
{
  public function get($model, $key, $value, $attributes)
  {
    $val = strtolower($value);
    switch($val)
    {
      case '':
      case '0':
      case 'off':
      case 'false':
      case ($val===false):
        return false;
        break;
      case '1':
      case 'on':
      case 'true':
      case ($val===true):
        return true;
        break;
    }
  }

  public function set($model, $key, $value, $attributes)
  {
    $val = strtolower($value);
    switch($val)
    {
      case '':
      case '0':
      case 'off':
      case 'false':
      case ($val===false):
        return 'off';
        break;
      case '1':
      case 'on':
      case 'true':
      case ($val===true):
        return 'on';
        break;
    }
  }
}
```

Nova model:

```
public function fields(Request $request)
    {
        return [
            Boolean::make('Variable', 'variable')->trueValue('on')->falseValue('off')->sortable()->filterable(),
```

If i try to filter for false value ('off') seems to work, but it doesnt return any result while set to true value (on)

I tried as well with a custom filter, a simple select on/off but that didnt worked too.
Issue is related to the Boolean type field, if i change it to text works fine.

**Comment by crynobone** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/172966/200427531-9ead7bd7-e808-4c82-99dc-52a6b1497d6d.png)

Since we can't 100% verify the correct query to be used for the filter you need to manually handle it. The filterable Boolean field should return `true` and `false` as `$value` and you need to handle the query manually.
